x/programs/cmd/simulator/cmd/plan.go

```diff
@@ -248,7 +251,11 @@ func runStepFunc(
 
 			return nil
 		}
-		id, response, balance, err := programExecuteFunc(ctx, log, db, params[0].Value.(ids.ID), params[1:], method, maxUnits)
+		id, err := ids.ToID(params[0].Value)
+		if err != nil {
+			return err
+		}
+		id, response, balance, err := programExecuteFunc(ctx, log, db, id, params[1:], method, maxUnits)
```

Comment:
remind me again why it's `param[1:]`?

Reply:
That's because within all params, the id of the program that is called is passed as the first of them.
The first parameter always has a special treatment and it is defined in this function that you tagged